bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveManufacturerSpecificCommandClass.java

```diff
@@ -83,6 +83,17 @@ public void handleApplicationCommandRequest(SerialMessage serialMessage,
 				logger.debug(String.format("NODE %d: Manufacturer ID = 0x%04x", this.getNode().getNodeId(), this.getNode().getManufacturer()));
 				logger.debug(String.format("NODE %d: Device Type = 0x%04x", this.getNode().getNodeId(), this.getNode().getDeviceType()));
 				logger.debug(String.format("NODE %d: Device ID = 0x%04x", this.getNode().getNodeId(), this.getNode().getDeviceId()));
+
+				// Check for Manufacturer Proprietary Command Class for this device.
+				final CommandClass commandClass = CommandClass.getCommandClass(0x91 + tempMan + tempDeviceType);
+				if (commandClass != null && getNode().getCommandClass(commandClass) == null){
+					logger.debug(String.format("NODE %d: Command class %s (0x%02x) not found, trying to add it.", getNode().getNodeId(), commandClass.getLabel(), commandClass.getKey()));
+					final ZWaveCommandClass zwaveCommandClass = ZWaveCommandClass.getInstance(commandClass.getKey(), getNode(), getController());
+					if (zwaveCommandClass != null) {
+						logger.debug(String.format("NODE %d: Adding command class %s (0x%02x)", getNode().getNodeId(), commandClass.getLabel(), commandClass.getKey()));
+						getNode().addCommandClass(zwaveCommandClass);
+					}
+				}
```

Comment:
I'm not sure why you needed to add this here - this isn't where most of the command classes get created. Only the mandatory classes are being created here, so I think this code can be removed and it will not matter....